### PR TITLE
BaSP-TG-76: Employee List Component

### PR DESCRIPTION
**Pull request by MartinDidoli**
- Employee Table List
- Delete / Edit Icon
- Delete feature
- Redirect to Add Form
- Warning Message for delete

### CODE REVIEW

File: src/Components/Employees/index.js [hunk 1]
```diff
@@ -1,27 +1,62 @@
...
+      console.error(error);
+    }
+  }, []);
 
-  useEffect(() => {
-    fetch(`https://jsonplaceholder.typicode.com/users`)
-      .then((response) => response.json())
-      .then((response) => {
-        saveEmployees(response);
+  const deleteItem = (_id) => {
```
Comment (marcomigliaro) [inline src/Components/Employees/index.js:18]: Should respect the structure. async (_id) and await fetch

File: src/Components/Employees/index.js [hunk 2]
```diff
@@ -1,27 +1,62 @@
...
-  }, []);
+      console.log(response);
+      alert('Employee deleted');
+    } catch (error) {
+      console.error(error);
+    }
+    setList(list.filter((listItem) => listItem._id !== _id));
+  };
+
+  /*const addItem = ({ firstName, lastName, phone, email, active }) => {
```
Comment (marcomigliaro) [inline src/Components/Employees/index.js:31]: Remove comments

File: src/Components/Employees/ListItem.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+const ListItem = ({ listItem, deleteItem /*editItem*/ }) => {
+  const handleDelete = (_id) => {
+    deleteItem(_id);
+  };
+
+  /*const handleEdit = (_id) => {
```
Comment (marcomigliaro) [inline src/Components/Employees/ListItem.jsx:8]: Remove comments

File: src/Components/Employees/ListItem.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+const ListItem = ({ listItem, deleteItem /*editItem*/ }) => {
+  const handleDelete = (_id) => {
```
Comment (marcomigliaro) [inline src/Components/Employees/ListItem.jsx:4]: We can avoid creating a new function for this case, using deleteItem() should work fine and replace it on the onClick

File: src/Components/Employees/ListItem.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    console.log('item', item);
+    return (
+      <tr key={item.id}>
+        <td>{item.firstName}</td>
+        <td>{item.lastName}</td>
+        <td>{item.phone}</td>
+        <td>{item.email}</td>
+        <td>{item.active.toString()}</td>
+        <td>
+          <a href="http://localhost:4000/employees/FormEmployee">
```
Comment (marcomigliaro) [inline src/Components/Employees/ListItem.jsx:22]: Shouldn´t use this, could delete this <a> and just leave the button for now. In the next ticket where edit feature is added you can focus on this

File: src/Components/Employees/AddItem.jsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react';
+import styles from './employees.module.css';
+
+const AddItem = ({ addItem }) => {
```
Comment (marcomigliaro) [inline src/Components/Employees/AddItem.jsx:4]: Would be better to change function name to EmployeeForm

File: src/Components/Employees/AddItem.jsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            <label>Last Name</label>
+            <input type="text" name="lastName" value={userInput.lastName} onChange={onChange} />
+          </div>
+          <div>
+            <label>Phone</label>
+            <input type="number" name="phone" value={userInput.phone} onChange={onChange} />
+          </div>
+          <div>
+            <label>Email</label>
+            <input type="text" name="email" value={userInput.email} onChange={onChange} />
```
Comment (marcomigliaro) [inline src/Components/Employees/AddItem.jsx:50]: Type should be email

File: src/Components/Employees/ListItem.jsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+const ListItem = ({ listItem, deleteItem }) => {
+  const test = listItem.map((item) => {
+    console.log('item', item);
```
Comment (cocolobos) [inline src/Components/Employees/ListItem.jsx:5]: remove console log

File: src/Components/Employees/ListItem.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+const ListItem = ({ listItem, deleteItem }) => {
+  const test = listItem.map((item) => {
```
Comment (cocolobos) [inline src/Components/Employees/ListItem.jsx:4]: function name should be descriptive

**Review by cocolobos — APPROVED** [review]

**Review by DeepFreq — APPROVED** [review]

**Pull request merged**